hyperspy/tests/io/test_hdf5.py

```diff
@@ -365,6 +368,315 @@ def teardown_method(self, method):
+        m = markers.point(x=x, y=y, color=color, size=size)
+        m.name = name
+        s.add_marker(m, permanent=True)
+        with tempfile.TemporaryDirectory() as tmp:
+            filename = tmp + '/testloadfile.hdf5'
+        s.save(filename)
+        s1 = load(filename)
+        assert s1.metadata.Markers.has_item(name)
+        m1 = s1.metadata.Markers.get_item(name)
+        assert m1.get_data_position('x1') == x
+        assert m1.get_data_position('y1') == y
+        assert m1.get_data_position('size') == size
+        assert m1.marker_properties['color'] == color
+        assert m1.name == name
+
+    def test_save_load_permanent_marker_all_types(self):
+        x1, y1, x2, y2 = 5, 2, 1, 8
+        s = Signal2D(np.arange(100).reshape(10, 10))
+        m0 = markers.point(x=x1, y=y1)
+        m1 = markers.horizontal_line(y=y1)
+        m2 = markers.horizontal_line_segment(x1=x1, x2=x2, y=y1)
+        m3 = markers.line_segment(x1=x1, x2=x2, y1=y1, y2=y2)
+        m4 = markers.rectangle(x1=x1, x2=x2, y1=y1, y2=y2)
+        m5 = markers.text(x=x1, y=y1, text="test")
+        m6 = markers.vertical_line(x=x1)
+        m7 = markers.vertical_line_segment(x=x1, y1=y1, y2=y2)
+
+        s.add_marker(m0, permanent=True)
+        s.add_marker(m1, permanent=True)
+        s.add_marker(m2, permanent=True)
+        s.add_marker(m3, permanent=True)
+        s.add_marker(m4, permanent=True)
+        s.add_marker(m5, permanent=True)
+        s.add_marker(m6, permanent=True)
+        s.add_marker(m7, permanent=True)
+
+        with tempfile.TemporaryDirectory() as tmp:
+            filename = tmp + '/testallmarkersfile.hdf5'
+        s.save(filename)
+        s1 = load(filename)
+        m_dict = s1.metadata.Markers
+
+        m0_new_dict = sanitize_dict(m_dict.get_item(m0.name)._to_dictionary())
+        m1_new_dict = sanitize_dict(m_dict.get_item(m1.name)._to_dictionary())
+        m2_new_dict = sanitize_dict(m_dict.get_item(m2.name)._to_dictionary())
+        m3_new_dict = sanitize_dict(m_dict.get_item(m3.name)._to_dictionary())
+        m4_new_dict = sanitize_dict(m_dict.get_item(m4.name)._to_dictionary())
+        m5_new_dict = sanitize_dict(m_dict.get_item(m5.name)._to_dictionary())
+        m6_new_dict = sanitize_dict(m_dict.get_item(m6.name)._to_dictionary())
+        m7_new_dict = sanitize_dict(m_dict.get_item(m7.name)._to_dictionary())
+
+        assert len(list(s1.metadata.Markers)) == 8
+        assert sanitize_dict(m0._to_dictionary()) == m0_new_dict
+        assert sanitize_dict(m1._to_dictionary()) == m1_new_dict
+        assert sanitize_dict(m2._to_dictionary()) == m2_new_dict
+        assert sanitize_dict(m3._to_dictionary()) == m3_new_dict
+        assert sanitize_dict(m4._to_dictionary()) == m4_new_dict
+        assert sanitize_dict(m5._to_dictionary()) == m5_new_dict
+        assert sanitize_dict(m6._to_dictionary()) == m6_new_dict
+        assert sanitize_dict(m7._to_dictionary()) == m7_new_dict
```

Comment:
I can see that you like copy-paste'ing ^^ 

I'd format this as a loop / iterator of some kind, but I guess it does not matter in the end